packages/react/src/components/Footer/Footer.js

```diff
@@ -6,8 +6,21 @@
  */
 
 import React from 'react';
-import { localeSelector } from '../../internal/FeatureFlags';
-import LocaleSelector from './LocaleSelector';
+import PropTypes from 'prop-types';
+// import { localeSelector } from '../../internal/FeatureFlags';
+// import LocaleSelector from './LocaleSelector';
+import { settings } from 'carbon-components';
+import classNames from 'classnames';
+import FooterTitle from './FooterTitle';
+import FooterNav from './FooterNav';
+import LegalNav from './LegalNav';
+
+import footerMenuData from './__stories__/data/footer-menu';
+import footerLegalData from './__stories__/data/footer-legal';
+
+console.log(footerLegalData);
+
+const { prefix } = settings;
 
 /** Footer component */
 class Footer extends React.Component {
```

Comment:
Thanks for these comments. It's been a while since I was in React last, and forgot there was a difference. The others we might be able to get away with (as long as children can still have state because the accordion from requires state I believe), but this one it really depends on how our adopters need it. If we are going to just allow adopters pass in whatever, and the with the UI Shell pass in the correct data then that's great. I was assuming the footer would make handle state and make the API call to translation by itself so we get the same data every time and not open that door for the incorrect formation of navigation items. In fact I was working on making that happen now, but maybe I'll pause before I get too far off track.

Reply:
@photodow check out how I handle this in MastheadSearch. I keep it as a stateless component, but use useReducer for handling the API changes with autosuggest. 

cc: @kennylam

Reply:
I'll check it out.